### ISSUE TITLE
Web: "Expire After" dropdown options in create share link modal hidden behind scrollbar

### ISSUE THREAD

**Issue by Snowknight26** [immich-app/immich#11984]
### The bug

When attempting to create a shared link to an asset/album, opening the "Expire after" dropdown shows the options such that scrolling is necessary to see them.

Probably a DOM element location issue or a z-index issue, though this dropdown seems to be a non-standard component or something.

How it current is:
![image](https://github.com/user-attachments/assets/926c728d-3949-46f1-b75f-27dd7df62eaa)

How it should be:
![image](https://github.com/user-attachments/assets/0a4f8bde-974b-44f6-98ad-0c1663b446b8)

### The OS that Immich Server is running on

Windows 10 22H2 (19045.4780)

### Version of Immich Server

v1.112.1

### Version of Immich Mobile App

N/A

### Platform with the issue

- [ ] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1. Navigate to Photos
2. Click on a photo
3. Click Share in the top nav bar
4. Click the "Expire after" dropdown
```

**Comment by ben-basten** [comment]
@alextran1502 what would you think about replacing this custom dropdown with a standard `select` component, like what's used in administration settings? This would fix the issue, along with allowing keyboard navigation and typing to search for an option.

![Screenshot 2024-08-26 at 07 43 24](https://github.com/user-attachments/assets/ba4c9d77-1604-4175-976b-86a1c1276681)

**Comment by bo0tzz** [comment]
I don't see why it should be a custom component, so yes please.

**Comment by stumpigit** [comment]
I have added the functionality for the combo box in the still open PR #11810 that the selection is automatically opened upwards if there is no more space in the scrollbar below. In my opinion, this works very well. If you are interested, I could also implement this for the dropdown button?

**Comment by ben-basten** [comment]
> If you are interested, I could also implement this for the dropdown button?

Thanks for the suggestion, @stumpigit! I'm actually thinking of removing this custom dropdown altogether in favor of using a native `select` element. This will fix the current issue, while also improving keyboard navigation. I have some code in progress with a fix.

**Comment by stumpigit** [comment]
Yes, it makes perfect sense to use a native select here. Thank you very much for the implementation!

For the comboboxes, I would just be happy if my PR could be looked at so that I could continue with further implementations (Star Rating filter).
